### PR TITLE
Work around ScopeCheck's Jank

### PR DESCRIPTION
**Pull request by CodaFi**
Fix support for rebinding arrows by breaking it a little.  Temporary workaround for applications in-between arrows by making them explicit.

### CODE REVIEW

File: Sources/Moho/ScopeCheck.swift [hunk 1]
```diff
@@ -87,51 +89,96 @@ extension NameBinding {
...
 
-      var args = [Expr]()
-      for e in syntax.exprs.dropFirst() {
-        let elim = self.scopeCheckExpr(e)
-        args.append(elim)
-      }
+      switch syntax.exprs[0] {
+      case let headExpr as NamedBasicExprSyntax:
+        let n = QualifiedName(ast: headExpr.name)
+        guard n.string != "->" else {
```
Comment (harlanhaskins) [inline Sources/Moho/ScopeCheck.swift:147]: Would be nice to add an `isArrow` property...

Comment (CodaFi) [reply Sources/Moho/ScopeCheck.swift:147]: It would be nice to delete this monstrosity and do it correctly...

Comment (harlanhaskins) [reply Sources/Moho/ScopeCheck.swift:147]: 👍👏👍👏👍👏👍👏👍👏

**Comment by CodaFi** [comment]
The precursor to better things...

⛵️

**Pull request merged**